wave_lang/kernel/wave/analysis/index_sequence_analysis.py

```diff
@@ -217,7 +217,7 @@ def verify_nodes(trace: CapturedTrace, constraints: list[Constraint]):
             continue
         if isinstance(custom.type, DataType):
             continue
-        assert custom.index, f"Index not set for node {custom.fx_node}: {custom}"
+        assert custom.index != None, f"Index not set for node {custom.fx_node}: {custom}"
```

Comment:
I'm concerned about this change.  There is a lit test that's hitting this with a SelectOp that wasn't before.  The index is an empty dictionary, which makes sense because in the case of that test it's a scalar.  But I worry that this could now be missing something useful that it used to catch.